R/readAxivity.R

```diff
@@ -485,11 +485,20 @@ readAxivity = function(filename, start = 0, end = 0, progressBar = FALSE, desire
     rawLast = prevLength + rawPos - 1
     rawTime[rawPos:rawLast] = time[1:prevLength]
     rawAccel[rawPos:rawLast,] = as.matrix(prevRaw$data)
-    # lastTime = time[prevLength]
+    
+    deltaTime = diff(rawTime[rawPos:rawLast])
     ###########################################################################
     # resampling of measurements
     last = pos + 200;
     if (pos + 200 > nr) last = nr
+    if (rawTime[rawLast] >= timeRes[last]) {
+      # there has been a time jump
+      # so, time jump needs to be adjusted for in last index
+      timejump = rawTime[rawLast] - timeRes[last]
+      positions2add = floor(timejump * prevRaw$frequency)
+      last = last + positions2add
```

Comment:
I think (theoretically) another jump could happen during the newly added positions, and so rawTime[rawLast] could still be > timeRes[last], even after this adjustment. This probably won't happen often, but maybe it's best to just take care of it.

Reply:
Thanks. How about we change the `if` statement to a `while` statement, would that address it? So:

```
if (rawTime[rawLast] > timeRes[last]) {
      # there has been a time jump
      # so, time jump needs to be adjusted for in last index
      timejump = rawTime[rawLast] - timeRes[last]
      positions2add = floor(timejump * prevRaw$frequency)
      last = last + positions2add
      if (last > nr) last = nr
}

```
would become:

```
while (rawTime[rawLast] > timeRes[last]) {
      # there has been a time jump
      # so, time jump needs to be adjusted for in last index
      timejump = rawTime[rawLast] - timeRes[last]
      positions2add = floor(timejump * prevRaw$frequency)
      last = last + positions2add
      if (last > nr) last = nr
}
```

Reply:
Maybe `while ((rawTime[rawLast] > timeRes[last]) & (last < nr))` ?
Otherwise I'm worried about getting stuck in an infinite loop if newly added positions push us past nr.

Reply:
I am actually not sure now this is needed.
`rawTime[rawLast]` is a constant in these 8 lines of the code.
`timeRes` is regularly sampled at the top and by that does not comes with unexpected jumps in time.
Therefore, incrementing the value of `last` should go reliably.

Reply:
Should I merge this PR, such that you can create a new PR in the upcoming weeks with your attempt to further improve the code?

Reply:
yes, sounds great!